tests/test_tensorflow_sparse_data_api.py

```diff
@@ -0,0 +1,243 @@
+    def test_tiledb_tf_sparse_data_api_with_sparse_data_dense_label(self):
+        for input_shape in INPUT_SHAPES:
+            with self.subTest():
+                model = create_model(
+                    input_shape=input_shape, num_of_classes=NUM_OF_CLASSES
+                )
+
+                array_uuid = str(uuid.uuid4())
+                tiledb_uri_x = os.path.join(self.get_temp_dir(), "x" + array_uuid)
+                tiledb_uri_y = os.path.join(self.get_temp_dir(), "y" + array_uuid)
+
+                dataset_shape_x = (ROWS, input_shape)
+                dataset_shape_y = (ROWS, NUM_OF_CLASSES)
+
+                ingest_in_tiledb_sparse(
+                    uri=tiledb_uri_x,
+                    data=create_sparse_array_one_hot_2d(dataset_shape_x),
+                    batch_size=BATCH_SIZE,
+                )
+                ingest_in_tiledb(
+                    uri=tiledb_uri_y,
+                    data=np.random.rand(*dataset_shape_y),
+                    batch_size=BATCH_SIZE,
+                )
+
+                with tiledb.SparseArray(tiledb_uri_x, mode="r") as x, tiledb.DenseArray(
+                    tiledb_uri_y, mode="r"
+                ) as y:
+                    tiledb_dataset = TensorflowTileDBSparseDataset(
+                        x_array=x, y_array=y, batch_size=BATCH_SIZE
+                    )
+
+                    self.assertIsInstance(tiledb_dataset, tf.data.Dataset)
+                    model.fit(tiledb_dataset, verbose=0, epochs=2)
+
+    @testing_utils.run_v2_only
+    def test_tiledb_tf_sparse_data_api_with_sparse_data_diff_number_of_batch_x_y_rows(
+        self,
+    ):
+        for input_shape in INPUT_SHAPES:
+            with self.subTest():
+                array_uuid = str(uuid.uuid4())
+                tiledb_uri_x = os.path.join(self.get_temp_dir(), "x" + array_uuid)
+                tiledb_uri_y = os.path.join(self.get_temp_dir(), "y" + array_uuid)
+
+                dataset_shape_x = (ROWS, input_shape)
+                dataset_shape_y = (ROWS, NUM_OF_CLASSES)
+
+                # Empty one random row
+                spoiled_data = create_sparse_array_one_hot_2d(dataset_shape_x)
+                spoiled_data[np.nonzero(spoiled_data[0])] = 0
+
+                ingest_in_tiledb_sparse(
+                    uri=tiledb_uri_x,
+                    data=spoiled_data,
+                    batch_size=BATCH_SIZE,
+                )
+                ingest_in_tiledb(
+                    uri=tiledb_uri_y,
+                    data=np.random.rand(*dataset_shape_y),
+                    batch_size=BATCH_SIZE,
+                )
+
+                with tiledb.SparseArray(tiledb_uri_x, mode="r") as x, tiledb.DenseArray(
+                    tiledb_uri_y, mode="r"
+                ) as y:
+                    _ = TensorflowTileDBSparseDataset(
+                        x_array=x, y_array=y, batch_size=BATCH_SIZE
+                    )
+                    self.assertRaises(Exception)
```

Comment:
Looks like this should be 
```
with self.assertRaises(Exception):
    TensorflowTileDBSparseDataset(x_array=x, y_array=y, batch_size=BATCH_SIZE)
```

Reply:
I tried that as well [here]( https://github.com/TileDB-Inc/TileDB-ML/pull/28/commits/281e3ea114f218772af46a60b4a376e6123655ee). Thinking the same. It turns out though that `from_generator` is not called during creation of the dataset, but instead when the dataset is used by an iterator in an online fashion. However it is a good catch since I was not testing that either: I will change to 
```
tiledb_dataset = TensorflowTileDBSparseDataset(x_array=x, y_array=y, batch_size=BATCH_SIZE)
    with self.assertRaises(Exception):
         model.fit(tiledb_dataset, verbose=0, epochs=2)
```

`from_generator` is being activated during `fit`